module/VuFindSearch/src/VuFindSearch/Backend/Blender/Backend.php

```diff
@@ -0,0 +1,590 @@
+    }
+
+    /**
+     * Retrieve a single document.
+     *
+     * @param string   $id     Document identifier
+     * @param ParamBag $params Search backend parameters
+     *
+     * @return \VuFindSearch\Response\RecordCollectionInterface
+     */
+    public function retrieve($id, ParamBag $params = null)
+    {
+        throw new \Exception('Blender does not support retrieve');
+    }
+
+    /**
+     * Return the record collection factory.
+     *
+     * Lazy loads a generic collection factory.
+     *
+     * @return RecordCollectionFactoryInterface
+     */
+    public function getRecordCollectionFactory()
+    {
+        return null;
+    }
+
+    /**
+     * Get active backends for a search
+     *
+     * @param ParamBag $params Search backend parameters
+     *
+     * @return array
+     */
+    protected function getActiveBackends(ParamBag $params): array
+    {
+        $activeBackends = $this->backends;
+
+        // Handle the blender_backend pseudo-filter
+        $fq = $params->get('fq') ?? [];
+        $filteredBackends = [];
+        foreach ($fq as $filter) {
+            $advanced = preg_match(
+                '/\{!tag=blender_backend_filter}blender_backend:\((.+)\)/',
+                $filter,
+                $matches
+            );
+            if ($advanced) {
+                $filter = explode(' OR ', $matches[1]);
+            }
+            foreach ((array)$filter as $current) {
+                if (strncmp($current, 'blender_backend:', 16) === 0) {
+                    $active = trim(substr($current, 16), '"');
+                    if (!isset($activeBackends[$active])) {
+                        throw new \Exception(
+                            "Invalid blender_backend filter: Backend $active not"
+                            . ' enabled'
+                        );
+                    }
+                    $filteredBackends[$active] = $activeBackends[$active];
+                }
+            }
+        }
+        if ($filteredBackends) {
+            $activeBackends = $filteredBackends;
+        }
+        foreach ($fq as $current) {
+            if (strncmp($current, '-blender_backend:', 17) === 0) {
+                $disabled = trim(substr($current, 17), '"');
+                if (isset($activeBackends[$disabled])) {
+                    unset($activeBackends[$disabled]);
+                }
+            }
+        }
+
+        return $activeBackends;
+    }
+
+    /**
+     * Get a record from the given backend by offset
+     *
+     * @param array $backendDetails Details for the backend
+     * @param array $backendRecords Record buffer
+     * @param int   $offset         Record offset
+     * @param int   $blockSize      Blending block size
+     *
+     * @return RecordInterface|null
+     */
+    protected function getRecord(
+        array $backendDetails,
+        array &$backendRecords,
+        int $offset,
+        int $blockSize
+    ): RecordInterface {
+        if (!$backendRecords) {
```

Comment:
Is it potentially misleading/confusing that the offset and block size are ignored if `$backendRecords` is already defined?